phonon/client.py

```diff
@@ -0,0 +1,42 @@
+import redis
+import zlib
+
+
+class ShardedClient(object):
+
+    key_mappings = {
+        'get': 0, 'hget': 0,
+        'set': 0, 'hset': 0,
+        'hgetall': 0,
+    }
+
+    def __init__(self, hosts=None, port=6379, db=0):
+        self.hosts = sorted(hosts)
+        self.clients = [redis.StrictRedis(host=host, port=port, db=db) for host in self.hosts]
+
+    def get_key(self, method, *args, **kwargs):
```

Comment:
We shard on the key, which could be dependent on the method. In the case of some of the multi- methods the key can be complicated. There's a mapping set up there to act as a placeholder for some potentially more complicated get-getting functionality. That key is what we shard on.